layers/+completion/spacemacs-helm/packages.el

```diff
@@ -180,7 +180,10 @@
 
       ;; use helm by default for M-x, C-x C-f, and C-x b
       (unless (configuration-layer/package-usedp 'smex)
-        (global-set-key (kbd "M-x") 'helm-M-x))
+        (global-set-key (kbd "M-x") 'helm-M-x)
+        (spacemacs/set-leader-keys
+          "SPC" 'helm-M-x)
```

Comment:
I'm not sure about this solution as it is discussed in #5999 that not everybody has this problem. But for sure `dotspacemacs-emacs-command-key` should be used instead of literal `"SPC"`.